File: app/src/main/java/com/elewise/ldmobile/ui/DocActivity.java
package com.elewise.ldmobile.ui;

import android.os.Bundle;
import com.google.android.material.tabs.TabLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;
import android.view.MenuItem;

import com.elewise.ldmobile.R;
import com.elewise.ldmobile.api.ParamDocumentDetailsResponse;
import com.elewise.ldmobile.service.Session;

import java.util.ArrayList;
import java.util.List;

public class DocActivity extends BaseActivity {

    private SectionsPagerAdapter mSectionsPagerAdapter;
    private ViewPager viewPager;
    private Session session;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_doc);

        session = Session.getInstance();

        mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());
        viewPager = findViewById(R.id.vpDoc);
        viewPager.setAdapter(mSectionsPagerAdapter);

        // Передаём ViewPager в TabLayout
        TabLayout tabLayout = findViewById(R.id.tlDocs);
        tabLayout.setupWithViewPager(viewPager);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        updateActionBar(session.getCurrentDocumentDetail().getDoc_title());
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == android.R.id.home) {
            finish();
            return true;
        } else {
            return super.onOptionsItemSelected(item);
        }
    }

    public class SectionsPagerAdapter extends FragmentPagerAdapter {

        private String tabTitles[];

        private String TAB_TITLE_DOCUMENT = getString(R.string.doc_activity_tab_document);
        private String TAB_TITLE_LINES = getString(R.string.doc_activity_tab_lines);
        private String TAB_TITLE_HISTORY = getString(R.string.doc_activity_tab_history);

        SectionsPagerAdapter(FragmentManager fm) {
            super(fm);

            List<String> listTitles = new ArrayList<>();

            ParamDocumentDetailsResponse curentDocument = session.getCurrentDocumentDetail();
            if (curentDocument.getHeader_attributes().length > 0) {
                listTitles.add(TAB_TITLE_DOCUMENT);
            }
            if (curentDocument.getLines() != null && curentDocument.getLines().length > 0) {
                listTitles.add(TAB_TITLE_LINES);
            }

            if (curentDocument.getHistory() != null && curentDocument.getHistory().length > 0) {
                listTitles.add(TAB_TITLE_HISTORY);
            }

            tabTitles = listTitles.toArray(new String[listTitles.size()]);
        }

        @Override
        public Fragment getItem(int position) {

            String name = tabTitles[position];
            if (name.equals(TAB_TITLE_DOCUMENT)) {
                return DocFragment.newInstance();
            } else if (name.equals(TAB_TITLE_LINES)) {
                return ItemsFragment.newInstance();
            } else {
                return HistoryFragment.newInstance();
            }
        }

        @Override
        public int getCount() {
            return tabTitles.length;
        }

        @Override public CharSequence getPageTitle(int position) {
            // генерируем заголовок в зависимости от позиции
            return tabTitles[position];
        }
    }
}

File: app/src/main/java/com/elewise/ldmobile/ui/DocActivity.kt
package com.elewise.ldmobile.ui

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.os.Environment
import android.provider.ContactsContract.Directory.PACKAGE_NAME
import android.view.MenuItem
import androidx.core.content.FileProvider
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.fragment.app.FragmentStatePagerAdapter
import com.elewise.ldmobile.BuildConfig
import com.elewise.ldmobile.R
import com.elewise.ldmobile.service.Session
import kotlinx.android.synthetic.main.activity_doc.*
import java.io.File


class DocActivity : BaseActivity() {
    private lateinit var session: Session

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_doc)
        session = Session.getInstance()
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        updateView()
    }


    private fun updateView() {
        val mSectionsPagerAdapter = SectionsPagerAdapter(supportFragmentManager)
        vpDoc.adapter = mSectionsPagerAdapter
        mSectionsPagerAdapter.notifyDataSetChanged()
        // Передаём ViewPager в TabLayout
        tlDocs.setupWithViewPager(vpDoc)
        updateActionBar(session.currentDocumentDetail.doc_title ?: "")
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        updateView()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (item.itemId == android.R.id.home) {
            finish()
            true
        } else {
            super.onOptionsItemSelected(item)
        }
    }

    inner class SectionsPagerAdapter internal constructor(fm: FragmentManager?) : FragmentStatePagerAdapter(fm) {
        private lateinit var tabTitles: ArrayList<String>
        private val TAB_TITLE_DOCUMENT = getString(R.string.doc_activity_tab_document)
        private val TAB_TITLE_LINES = getString(R.string.doc_activity_tab_lines)
        private val TAB_TITLE_HISTORY = getString(R.string.doc_activity_tab_history)

        override fun getItemPosition(`object`: Any): Int {
            return POSITION_NONE
        }

        override fun getItem(position: Int): Fragment {
            val name = tabTitles[position]
            return if (name == TAB_TITLE_DOCUMENT) {
                DocFragment()
            } else if (name == TAB_TITLE_LINES) {
                ItemsFragment.newInstance()
            } else {
                HistoryFragment.newInstance()
            }
        }

        override fun getCount(): Int {
            return tabTitles.size
        }

        override fun getPageTitle(position: Int): CharSequence? {
            // генерируем заголовок в зависимости от позиции
            return tabTitles[position]
        }

        init {
            val listTitles: MutableList<String> = ArrayList()
            with(session.currentDocumentDetail) {
                header_attributes?.let {
                    if (it.size > 0) {
                        listTitles.add(TAB_TITLE_DOCUMENT)
                    }
                    if (lines != null && lines.size > 0) {
                        listTitles.add(TAB_TITLE_LINES)
                    }
                    if (history != null && history.size > 0) {
                        listTitles.add(TAB_TITLE_HISTORY)
                    }
                    tabTitles = listTitles as ArrayList<String>
                } ?: run { tabTitles = ArrayList() }
            }
        }
    }
}